google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploy.java

```diff
@@ -97,16 +98,22 @@ public void deploy(
     if (environment.isStandard() || environment.isFlexCompat()) {
       appYamlName = FlexibleFacetEditor.APP_YAML_FILE_NAME;
     } else {
+      String moduleName = deploymentConfiguration.getModuleName();
```

Comment:
I know it looks like I'm duplicating this logic sequence a few times, but in each case the errors are handled differently. It's possible to refactor this but I'm not sure if it will be any cleaner.